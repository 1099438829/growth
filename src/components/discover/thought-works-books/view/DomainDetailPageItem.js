import React, { Component, PropTypes } from 'react';
import { Text, View } from 'react-native';
import AppStyle from '../../../../theme/styles';
import Line from '../../../Line';

class DomainDetailPageItem extends Component {
  static componentName = 'DomainDetailPageItem';

  static propTypes = {
    content:
        PropTypes.shape(
          PropTypes.string.isRequired,
          PropTypes.arrayOf(PropTypes.shape(
            PropTypes.string.isRequired,
            ),
          )),
  };

  static defaultProps = {
    content: {},
  };

  constructor(props) {
    super(props);
    this.state = {
      name: this.props.content.name,
      rowData: Array.from(new Array(this.props.content.books.length))
        .map((val, index) => (this.props.content.books[index])) };
  }

  render() {
    const rows = this.state.rowData.map((val, index) => (
      <View style={{ backgroundColor: 'white' }} key={val.concat(index)}>
        <View style={AppStyle.projectListContentStyle}>
          <Text>{val}</Text>
        </View>
        <Line />
      </View>
     ));
    return (
      <View>
        <View style={AppStyle.titleVerticalCenterColorGrayStyle}>
          <Text style={{ color: '#999' }}>{this.state.name}</Text>
        </View>
        {rows}
      </View>
    );
  }
}
export default DomainDetailPageItem;

