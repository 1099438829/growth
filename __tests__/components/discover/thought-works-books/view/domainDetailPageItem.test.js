import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import DomainDetailPageItem from '../../../../../src/components/discover/thought-works-books/view/DomainDetailPageItem';
import TWBOOKS from '../../../../../src/constants/TWBOOKS';

it('renders correctly', () => {
  const domainDetailPageItem = renderer.create(
    <DomainDetailPageItem
      content={{ name: '基础篇', books: TWBOOKS['zh-cn'].practical.books.basic }}
    />,
  );
  const domainDetailPageItemJson = domainDetailPageItem.toJSON();
  expect(domainDetailPageItemJson).toMatchSnapshot();
});
