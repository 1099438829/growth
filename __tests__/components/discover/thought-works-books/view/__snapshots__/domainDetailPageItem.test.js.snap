// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "rgb(245,245,245)",
        "height": 40,
        "justifyContent": "center",
        "paddingLeft": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999",
        }
      }
    >
      基础篇
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        代码整洁之道
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(240, 240, 240, .9)",
          "height": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        单元测试之道
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(240, 240, 240, .9)",
          "height": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        卓有成效的程序员
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(240, 240, 240, .9)",
          "height": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        测试驱动开发
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(240, 240, 240, .9)",
          "height": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        程序员的职业修养
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(240, 240, 240, .9)",
          "height": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        编写可读代码的艺术
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(240, 240, 240, .9)",
          "height": 1,
        }
      }
    />
  </View>
</View>
`;
